source/patterns/@aws-solutions-constructs/core/test/test-helper.ts

```diff
@@ -146,3 +147,23 @@ export function CreateTestCache(scope: Construct, id: string, vpc: ec2.IVpc, por
   newCache.addDependsOn(subnetGroup);
   return newCache;
 }
+
+/**
+ * Returns the Logical ID for the resource with a matching description.
+ *
+ * @param stack The CloudFormation Stack that contains the resource to get.
+ * @param resourceType The type of CloudFormation Resource, e.g., `AWS::KMS::Key`, `AWS::SNS::Topic`
+ * @param description The value of the Description property on the CloudFormation Resource
+ * @returns The Logical ID of the found resource
+ */
+export function getResourceLogicalIdFromDescription(stack: Stack, resourceType: string, description: string): string {
```

Comment:
Splitting the execution of these tests between this function and then checking the resulting logical ID in the test itself hides the extra level of confirmation. Let's expand this function to perform the entire "Is the resource we expect actually attached to the correct parent resource" check in one place:

```
function expectKmsKeyAttachedToCorrectResource(
  stack: Stack,
  parentResourceType: string,
  keyDescription: string) {

    // Find the unique KMS key we expect to be in
    // in the stack
    const template = Template.fromStack(stack);
    const resource = template.findResources('AWS::KMS::Key', {
      Properties: {
        Description: Match.exact(keyDescription)
      }
    });
    const [     const [ keyResourceId ] = Object.keys(resource); ] = Object.keys(resource);
  
    // Confirm that the key is attached to the
    // parent resource we expect
    expect(stack).toHaveResource(parentResourceType, {
      KmsMasterKeyId: {
        "Fn::GetAtt": [
          keyResourceId,
          "Arn"
        ]
      }
    });
   
  }
```

Reply:
Good suggestion - Updated in latest revision.